src/core/clipboardmanager.cpp

```diff
@@ -181,3 +183,56 @@ QgsFeature ClipboardManager::pasteFeatureFromClipboard()
 
   return feature;
 }
+
+bool ClipboardManager::pasteAsNewFeatureFromClipboardIntoLayer( QgsVectorLayer *layer )
+{
+  if ( !layer )
+  {
+    QgsMessageLog::logMessage( tr( "Paste failed: no destination layer provided" ), "QField" );
```

Comment:
Don't put that into the message log, that'll show up to strongly in the UI, just use qInfo() here.